packages/dom/src/dom/input-field-has-uncollapsed-selection.js

```diff
@@ -18,9 +18,25 @@ import isNumberInput from './is-number-input';
  * @return {boolean} Whether the input/textareaa element has some "selection".
  */
 export default function inputFieldHasUncollapsedSelection( element ) {
-	if ( ! isTextField( element ) && ! isNumberInput( element ) ) {
+	if ( ! isTextField( element ) && ! isHTMLInputElement( element ) ) {
```

Comment:
How about creating a new util here(instead of `isHTMLInputElement`) that wraps the checks for the specific inputs we want to let browsers handle it. I think these would be `number, email and time`? In that case we would add an extra same check below, if there is no selection but this is true.

Now if we have any other inputs like button, etc.. would return true and I don't think we need to do this.

Reply:
I'm not sure I understood your question; could you show me?

Reply:
In several places in the code base we use the `isNumberInput` for the exact same reason of this PR: trying also include `input[type="number"]` for some selection related actions/calculations. What I suggest is to make a new util that will include `isNumberInput` and similar checks as if we had `isEmailInput and isTimeInput`. Makes sense?

Reply:
Thanks, I understand the point now. What's interesting is that, the more I look at this PR and at existing usage of `isNumberInput`, the more I'm confused. :) I'll comment in more detail in the general thread.